File: sorting/InsertionSort.java
package JAVA.sorting;

public class InsertionSort {
    static int[] array = new int[]{4,3,2,1,5,8,6,9,7};



    public static void main(String[] args) {
        insertionSort(array);
        for(int data:array){
            System.out.println(data);
        }
    }

    static void insertionSort(int [] array){
        for(int i=1;i<array.length;i++){
            int key = array[i]; // 4
            int j = i-1;
            while(j >= 0 && array[j] > key){
                array[j+1] = array[j];
                j--;
            }
            array[j+1] =key;
        }
    }
}

File: GO-CODE/InsertionSort.go
package main

import "fmt"

func main(){

	array := []int{3,1,2,6,4,5,9,8,7}


	for i := 1; i<len(array);i++ {
		var data = array[i] // 2
		var j = i - 1 // 1

		

		for j >= 0 && array[j] > data { // 5 > 2
			array[i] = array[j]
			array[j] =data
		}

		fmt.Println(array)
	}
	
}